packages/compass-generative-ai/src/components/ai-optin-modal.tsx

```diff
@@ -149,8 +163,74 @@ export const AIOptInModal: React.FunctionComponent<OptInModalProps> = ({
           </Link>{' '}
           for more information.
         </div>
-      </Body>
-    </ConfirmationModal>
+      }
+    >
+      {isCloudOptIn && (
+        <Body className={bodyStyles}>
+          AI-powered features in Data Explorer supply users with an intelligent
+          toolset to build faster and smarter with MongoDB.
+          <Banner
+            data-testid="ai-optin-cloud-banner"
+            variant={isProjectAIEnabled ? 'info' : 'warning'}
+            className={bannerStyles}
+          >
+            {(() => {
```

Comment:
Nit: hard to read this lengthy IIFE inside a component, let's move it to a separate component instead if you want to isolate this logic?